### PR TITLE
update UI

### PR DESCRIPTION
**Pull request by benfpv**
- update UI buttons (when clicked) to lead to its respective function under main.
- main.currentUser is now established again.

### CODE REVIEW

File: ui/__pycache__/gameButton.cpython-312.pyc
Comment (AndrewJiangB) [inline ui/__pycache__/gameButton.cpython-312.pyc:1]: Don't include these in commit

File: ui/__pycache__/userInterface.cpython-312.pyc
Comment (AndrewJiangB) [inline ui/__pycache__/userInterface.cpython-312.pyc:1]: Don't include these in commit

File: ui/userInterface.py [hunk 1]
```diff
@@ -73,7 +82,8 @@ def check_mouseclick(self, mouse):
         return False, buttonIndex
     
 class Homepage:
-    def __init__(self, game, width, height, smallfont, color, color_light, color_hover):
+    def __init__(self, game, width, height, smallfont, color, color_light, color_hover, name = ""):
+        self.name = "homepage"
```
Comment (AndrewJiangB) [inline ui/userInterface.py:86]: lol what is the point of this

Comment (benfpv) [reply ui/userInterface.py:86]: I think the purpose of  name in __init__ was to have the calls for this generalizable to either future scenario: (1) we need to declare new similar class, or (2) use same class, for each new subordinate page (to homepage).
Line 86 is just placeholder lol.

Comment (benfpv) [reply ui/userInterface.py]: Removed the placeholder in newest push.

**Comment by benfpv** [comment]
### ALL REVIEW COMMENTS RESOLVED ###

**Comment by benfpv** [comment]
### @AndrewJiangB PUSH TO MAIN?!??!?!??!??!??!?!!??!?!?! ###

**Pull request merged**